app/src/main/java/org/simple/clinic/user/UserSession.kt

```diff
@@ -101,7 +102,25 @@ class UserSession @Inject constructor(
                 )
               }
         }
-        .flatMapCompletable { storeUser(it) }
+        .flatMapCompletable {
+          // TODO: also clear ongoing registration entry?
+          storeUser(it)
+        }
+  }
+
+  fun findExistingUser(phoneNumber: String): Single<FindUserResult> {
+    return registrationApi.findUser(phoneNumber)
+        .map<FindUserResult> { user -> FindUserResult.Found(user) }
```

Comment:
Opinion: I prefer to leave the generic parameter out of the operator by doing something like 
```kotlin 
map { FindUserResult.Found(it) as FindUserResult }
```

Reply:
Looks better, let's use this. I dislike keeping type information on operators too.

Reply:
Updated all usages in `UserSession` class